app/components/utiliti.tsx

```diff
@@ -10,7 +10,7 @@ import ContentWrapper from "~/components/content-wrapper";
 interface Props<T> {
   readonly label: string;
   // thought about using a `Map` here since it guarantees order, but we give up "DX"
-  readonly actions: Record<string, (input: string) => Promise<T>>;
+  readonly actions: Record<string, (input: string) => string | Object>;
```

Comment:
This is likely not a good idea because a few utilities will want to provide asynchronous output. I might checkout your PR and see if I can help you turn this back into a `Promise<T>`.